codegen/kuberesource/crd.go

```diff
@@ -117,27 +138,44 @@ func validateStructural(s *apiextv1.JSONSchemaProps) error {
 }
 
 func CustomResourceDefinition(
-	resource model.Resource,
-	validationSchema *apiextv1.CustomResourceValidation,
+	resources []model.Resource,
+	validationSchemas map[string]*apiextv1.CustomResourceValidation,
 	withoutSpecHash bool,
 ) *apiextv1.CustomResourceDefinition {
 
-	group := resource.Group.Group
-	version := resource.Group.Version
-	kind := resource.Kind
+	group := resources[0].Group.Group
+	kind := resources[0].Kind
 	kindLowerPlural := strings.ToLower(stringutils.Pluralize(kind))
 	kindLower := strings.ToLower(kind)
 
-	var status *apiextv1.CustomResourceSubresourceStatus
-	if resource.Status != nil {
-		status = &apiextv1.CustomResourceSubresourceStatus{}
-	}
-
 	scope := apiextv1.NamespaceScoped
-	if resource.ClusterScoped {
+	// TODO (dmitri-d) Should we check if all resources have the same scope
+	// and bail if not?
+	// Same for ShortNames and Categories below
+	if resources[0].ClusterScoped {
 		scope = apiextv1.ClusterScoped
 	}
```

Comment:
Its not like we do this today but yeah I would bail here

Reply:
Done